app/models/nic/bmc.rb

```diff
@@ -39,6 +37,24 @@ def ensure_physical
     def enc_attributes
       @enc_attributes ||= (super + %w(username password provider))
     end
+
+    private
+
+    def bmc_proxy
+      if subnet.present?
+        proxy = subnet.proxies.find { |subnet_proxy| subnet_proxy.has_feature?('BMC') }
+      end
+      proxy ||= SmartProxy.with_features("BMC").first
```

Comment:
Default scoping will apply taxonomies, so this could prevent saving of a host if a smart proxy isn't visible to one particular user.  Should this be unscoped, or scoped specifically to the same as the host?